src/lib/functions/runtimes/go/index.js

```diff
@@ -37,11 +40,32 @@ const checkGoInstallation = async ({ cwd }) => {
   }
 }
 
+const parseForSchedule = async ({ cwd, functionName }) => {
+  const manifestPath = await findUp('netlify.toml', { cwd, type: 'file' })
+  const manifest = await readFile(manifestPath, 'utf-8')
+  const { functions } = toml.parse(manifest)
+
+  const scheduledFunc = functions[functionName]
+  return scheduledFunc && scheduledFunc.schedule
+}
+
 const getBuildFunction = ({ func }) => {
   const functionDirectory = dirname(func.mainFile)
   const binaryPath = tempy.file(isWindows ? { extension: 'exe' } : undefined)
 
-  return () => build({ binaryPath, functionDirectory })
+  return async () => {
+    // From the current function directory, we look up parent directories for the netlify.toml file.
+    // If we find one, we parse it to see if a schedule is defined for the current function. It's okay
+    // if we don't find any, we just assume the function is not scheduled (i.e. undefined).
+    const schedule = await parseForSchedule({ cwd: functionDirectory, functionName: func.name })
```

Comment:
i don't like that this loads the file everytime even though the file was already loaded before.

i can see that `FunctionRegistry` calls `NetlifyFunction.build` and has the config already present.
consider passing the config down into this method instead of parsing the file anew.